### PR TITLE
R7081 realistic charge distribution and bug fixes

### PR DESCRIPTION
**Pull request by jcaravaca**
Added the charge distribution for the R7081 tubes as measured at Berkeley to have a realistic reference.
Fix bug that prevented from orient the PMTs in the simulation towards a fixed point.
Fix bug in Gsim.cc: when creating dark hits for several PMTs the code would crash.

### CODE REVIEW

File: src/geo/GeoPMTFactoryBase.cc [hunk 1]
```diff
@@ -136,6 +136,9 @@ G4VPhysicalVolume *GeoPMTFactoryBase::ConstructPMTs(DBLinkPtr table,
     dir_y = lpos_table->GetDArray("dir_y");
     dir_z = lpos_table->GetDArray("dir_z");
   } else {
+    dir_x = lpos_table->GetDArray("x"); //dummy values needed for pmtinfo.AddPMT
+    dir_y = lpos_table->GetDArray("x");
+    dir_z = lpos_table->GetDArray("x");
```
Comment (mastbaum) [inline src/geo/GeoPMTFactoryBase.cc]: I think this might be confusing to people reading the PMTInfo since there is no indication that the direction is in fact unknown. Ideally we could set the actual, calculated directions in PMTInfo, or failing that, set them to some obvious flag value.

File: src/core/Gsim.cc [hunk 2]
```diff
@@ -480,8 +480,8 @@ void Gsim::MakeEvent(const G4Event* g4ev, DS::Root* ds) {
 
   // Get the PMT type for IDPMTs. Then in the loop,
   // increment numPE only when the PE is in an IDPMT.
-  // Map from PMT ID numbers to objects for use later in noise calculation
-  std::map<int, DS::MCPMT*> mcpmtObjects;
+  //Map ID-INDEX for later noise calculation
+  std::map<int, int> mcpmtObjects;
```
Comment (mastbaum) [inline src/core/Gsim.cc]: Why this change?

File: src/core/Gsim.cc [hunk 3]
```diff
@@ -490,7 +490,7 @@ void Gsim::MakeEvent(const G4Event* g4ev, DS::Root* ds) {
     // Create and initialize a RAT DS::MCPMT 
     // note that GLG4HitPMTs are given IDs which are their index
     DS::MCPMT* rat_mcpmt = mc->AddNewMCPMT();
-    mcpmtObjects[a_pmt->GetID()] = rat_mcpmt;
+    mcpmtObjects[a_pmt->GetID()] = mc->GetMCPMTCount()-1; //at this point the size represent the index
```
Comment (mastbaum) [inline src/core/Gsim.cc]: This comment may be confusing -- this value is the index of the last element, rather than the size.

File: src/core/Gsim.cc (hunk 2 above)
Comment (mastbaum) [inline src/core/Gsim.cc]: Ah, I just saw the comment. If there's a situation where the `rat_mcpmt` doesn't exist here, we should understand what causes that -- probably aborting is still the right thing to do!

Comment (jcaravaca) [inline src/core/Gsim.cc]: The real problem is that we were defining pointers to the elements of the MCPMT std::vector. This is fine if you are not going to resize your vector. The issue arise when adding new PMTs due to dark hits, then the vector is relocated in memory and the pointers meaningless... I notice this problem in my simulation and the fix should work well.

Comment (mastbaum) [inline src/core/Gsim.cc]: Ah, I see! Great catch!

**Comment by mastbaum** [comment]
Thanks, @jcaravaca -- is this now good to merge?

**Comment by jcaravaca** [comment]
@mastbaum, your proposal makes sense to me. Do you prefer I include that instead in this pull request?

**Comment by mastbaum** [comment]
No, that's okay, they shouldn't conflict. Otherwise this is good to go?

**Comment by jcaravaca** [comment]
Yes, this is OK now
